### PR TITLE
Feature/confirm enable modal

### PR DESCRIPTION
**Pull request by danoswaltCL**
This add the enable confirm modal when clicking the details page top-most toggle in the action buttons.

This modal will close when the request is successful and the switch will then animate to move to enabled.

Disabling is nearly identical, but for review's sake this is plenty for one PR, I'll put up the disable status modal PR shortly after this is merged.

Some details like using the en.json translate properly and polishing up the styling are still needed, but trying to get this in for demo.

### CODE REVIEW

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts [hunk 1]
```diff
@@ -29,50 +28,49 @@ import { DialogService } from '../../../../../../../shared/services/common-dialo
...
+
+    if (slideToggleEvent.checked) {
+      this.openEnableConfirmModel();
+    } else {
+      // this.openDisableConfirmModel();
+      console.log('disable!');
+    }
+
+    // Note: we don't want the toggle to change state immediately because we have to pop a confirmation modal first, so we need override the default and flip it back
+    slideToggleEvent.checked = !slideToggleEvent.checked;
```
Comment (Yagnik56) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-details-page/feature-flag-details-page-content/feature-flag-overview-details-section-card/feature-flag-overview-details-section-card.component.ts:62]: I tried to disable the toggle, but it isn't working. can you confirm is it right behaviour?

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/enable-feature-flag-modal/enable-feature-flag-modal.component.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  ) {}
+
+  ngOnInit() {
+    this.listenForFeatureFlagStatusChanges();
+  }
+
+  listenForFeatureFlagStatusChanges(): void {
+    this.subscriptions = this.selectedFlagStatusChange$.subscribe(() => {
+      this.closeModal();
+    });
```
Comment (Yagnik56) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/enable-feature-flag-modal/enable-feature-flag-modal.component.ts:45]: why add a subscriber to close the modal, you can add a close modal at the end of `primaryButtonClick` function. Are we waiting for a status change only after that do you want to change status? It doesn't seem necessary, there won't be much delay there.

**Review by Yagnik56 — CHANGES_REQUESTED** [review]
There few doubts and suggestions @danoswaltCL, can you have a look?

File: frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.reducer.ts [hunk 3]
```diff
@@ -69,6 +71,21 @@ const reducer = createReducer(
...
+    isLoadingUpdateFeatureFlagStatus: true,
+  })),
+  on(FeatureFlagsActions.actionEnableFeatureFlagSuccess, (state, { response }) => {
+    const flag = response[0];
+    return adapter.updateOne(
+      { id: flag?.id, changes: { status: flag?.status } },
+      { ...state, isLoadingUpdateFeatureFlagStatus: false }
+    );
+  }),
+  on(FeatureFlagsActions.actionEnableFeatureFlagFailure, (state) => ({
```
[the remark this answers is not in the document]
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.reducer.ts:85]: disable isn't supposed to be part of this pr, i'll remove it.

File: frontend/projects/upgrade/src/app/core/feature-flags/feature-flags.service.ts [hunk 4]
```diff
@@ -27,17 +29,22 @@ export class FeatureFlagsService {
 
   isInitialFeatureFlagsLoading$ = this.store$.pipe(select(selectHasInitialFeatureFlagsDataLoaded));
   isLoadingFeatureFlags$ = this.store$.pipe(select(selectIsLoadingFeatureFlags));
+  isLoadingUpdateFeatureFlagStatus$ = this.store$.pipe(select(selectIsLoadingUpdateFeatureFlagStatus));
```
[the remark this answers is not in the document]
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/core/feature-flags/feature-flags.service.ts:32]: yes

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/enable-feature-flag-modal/enable-feature-flag-modal.component.ts (hunk 2 above)
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/enable-feature-flag-modal/enable-feature-flag-modal.component.ts:45]: it is a requirement that we handle all loading states. a front-end developer cannot say "it's probably going to be quick". The pattern for user-initiated http requests is that we don't move on until the response is received.

**Review by Yagnik56 — APPROVED** [review]

**Pull request merged**